src/workerd/server/server.c++

```diff
@@ -706,10 +708,10 @@ class Server::ExternalHttpService final: public Service, private kj::TaskSet::Er
       wrappedResponse = response;
       if (parent.rewriter->needsRewriteRequest()) {
         auto rewrite = parent.rewriter->rewriteOutgoingRequest(url, headers, metadata.cfBlobJson);
-        return parent.serviceAdapter->request(method, url, *rewrite.headers, requestBody, *this)
-            .attach(kj::mv(rewrite));
+        co_return co_await parent.serviceAdapter->request(
```

Comment:
Was there a reason for switching from `return` to `co_return co_await`?

Reply:
coroutines for life, never return promises, lifetimes get confusing and attachments become annoying.